clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -361,13 +365,21 @@ private void pollSelectionKeys(Iterable<SelectionKey> selectionKeys,
 
             } catch (Exception e) {
                 String desc = channel.socketDescription();
-                if (e instanceof IOException)
+                if (e instanceof ConnectException) {
+                    if (LogRateLimiter.warn(log, "Cannot connect to {}", desc, e,
+                            loggerCount, loggerCountMax))
+                        loggerCount++;
```

Comment:
Is this doing what you expect it to? You're returning `true` from the `LogRateLimiter` when the `count > max`. But that won't be true early on, which means you'll keep resetting the `loggerCount` to `0`. Won't this just keep the `loggerCount` at `0` permanently?